### PR TITLE
Enable master link replacement feature in .NET

### PR DESCRIPTION
**Pull request by sima-zhu**
This is a follow up step of master link replacement work.

Tested in template:
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=531026&view=logs&j=ceb84d66-6af3-5d95-9875-cd812f26b59e&t=bb6510a9-1dd3-5240-6b4b-d1bd7663119b

Retrieved release tag successfully:
image

Doc.Ms:
image

Github IO:
https://azuresdkdocs.blob.core.windows.net/$web/dotnet/Azure.Template/1.0.3-beta.10/index.html

### CODE REVIEW

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml [hunk 1]
```diff
@@ -24,12 +24,6 @@ jobs:
         displayName: 'Use Python 3.6'
         inputs:
           versionSpec: '3.6'
-      - template: /eng/common/pipelines/templates/steps/replace-relative-links.yml
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/archetype-sdk-client.yml:27]: This needs to wait until we fix the relative links.

File: eng/pipelines/templates/stages/archetype-net-release.yml [hunk 2]
```diff
@@ -230,6 +231,7 @@ stages:
                           BlobSASKey: '$(azure-sdk-docs-prod-sas)'
                           BlobName: '$(azure-sdk-docs-prod-blob-name)'
                           TargetLanguage: 'dotnet'
+                          ArtifactLocation: '$(Pipeline.Workspace)/${{artifact.safeName}}/packages'
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/archetype-net-release.yml:234]: Does this need to include the "packages" folder in the path?

File: sdk/template/Azure.Template/CHANGELOG.md [hunk 3]
```diff
@@ -1,7 +1,9 @@
 # Release History
+## 1.0.3-beta.9 (2020-09-09)
+- Enable the master link replacement feature, attemp 2
 
 ## 1.0.3-beta.8 (2020-09-09)
-- Enable the master link replacement feature
+- Enable the master link replacement feature, attemp 1
```
Comment (weshaggard) [inline sdk/template/Azure.Template/CHANGELOG.md:6]: nit: attemp -> attempt

Comment (sima-zhu) [reply sdk/template/Azure.Template/CHANGELOG.md:9]: Will fix the typo

**Review by weshaggard — APPROVED** [review]

**Pull request merged**